autoscale/src/main/java/com/sequenceiq/periscope/controller/DistroXAutoScaleClusterV1Controller.java

```diff
@@ -135,7 +135,8 @@ private DistroXAutoscaleClusterResponse updateClusterAutoScaleState(Cluster clus
 
     private DistroXAutoscaleClusterResponse updateClusterAutoScaleConfig(Cluster cluster,
             DistroXAutoscaleClusterRequest autoscaleClusterRequest) {
-
+        alertValidator.validateScheduleWithStopStart(cluster, autoscaleClusterRequest);
+        autoscaleClusterRequest.setUseStopStartMechanism(asClusterCommonService.enableStopStartBasedOnEntitlement(cluster, autoscaleClusterRequest));
```

Comment:
asClusterCommonService.setStopStartScalingState is already invoked further down.

It would be better to to perform these checks further down instead of modifying the incoming request.